aip/general/0008.md

```diff
@@ -7,22 +7,50 @@ placement:
   order: 40
 ---
 
-# AIP Style guide
+# AIP Style and Guidance
 
 AIP stands for **API Improvement Proposal**, which is a design document
-providing high-level, concise documentation for API development. The goal is for
-these documents to serve as the source of truth for API-related documentation at
-Google and the way API teams discuss and come to consensus on API guidance.
+providing high-level, concise documentation for API design and development. The
+goal is for these documents to serve as the source of truth for API-related
+documentation at Google and the way API teams discuss and come to consensus on
+API guidance.
 
 AIPs are most useful when they are clear and concise, and cover a single topic
 or inquiry well. In the same way that AIPs describe consistent patterns and
 style for use in APIs, they also _follow_ consistent patterns and style.
 
 ## Guidance
 
-AIPs **must** cover a single, discrete topic, and **should** fundamentally
-answer the question, "What do I do?" with actionable guidance. AIPs **may** also
-cover what _not_ to do, but **should not** cover _only_ anti-patterns.
+- AIPs **must** cover a single, discrete topic, and provide clear, actionable
+  guidance.
+- AIPs **must not** duplicate or contradict guidance in another AIP.
+- AIPs **may** also cover what _not_ to do, but **should not** cover _only_
+  anti-patterns.
+- If AIP guidance is conditional (e.g. a design pattern such as Jobs), the
+  guidance **must** clearly explain under what conditions the guidance should be
+  followed.
+
+AIPs **must** contain API guidance that provide a clear benefit to one or more
+types of clients or their authors, including but not limited to:
```

Comment:
Can you include an example (existing or otherwise) of what this looks like? Feels a bit nebulous to me how this will manifest in an AIP. Is this a new subsection in Rationale? I'd like to be able to point at something that fulfills this requirement, rather than debate what content does or does not address it.

Reply:
sorry for the delay. I added examples, can you see if that helps?

Reply:
Ok thanks. Last clarification, is this guidance (that you are adding with the `**must**`) to say that an AIP **must** spell out how it is beneficial to (a) clients? Or is this more of a meta point to say that "if the proposed guidance is cumbersome in these ways, it won't be accepted"?

Basically, how is this guidance here:
`AIPs **must** contain API guidance that provide a clear`

actionable in AIP text if at all

Reply:
I read it as "an AIP must provide a clear benefit to a client or user". I wasn't intending to even say it has to spell it out. 

It's the same level of ambiguity as "cover a single, discrete topic": that's somewhat subjective, but maybe clear enough to most that one can action on it (e.g. you say "split this up into separate AIPs").

In the case of this wording, the action would be "it's not clear that this guidance improves clients or users".

But yeah this could be read differently... how about "must contain API guidance that benefits one or more"... just getting rid of the "clear" part. 

What do you think? open to ideas too, assuming my intention is clear and you agree.

Reply:
I see what you mean! I see a requirement word and I automatically think about how the guidance would manifest concretely, so might just be me.

Idk why but the following reads better to me, but I'm not picky and will approve this as is:
```md
Guidance contained within an AIP **must** be beneficial to one or more
types of clients or their authors, including but not limited to:
```

Reply:
updated! thanks for the brainstorming.